### PR TITLE
Moves more steps of the release process to CI

### PR DESCRIPTION
**Pull request by vegaro**
Moves the following to CI:
- Making a GitHub release
- Opening PR with SNAPSHOT version

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -58,6 +58,34 @@ commands:
             echo $GPG_SIGNING_KEY_PW_NEW | gpg --batch --pinentry-mode=loopback --passphrase-fd 0 \
             --export-secret-keys > /home/circleci/.gnupg/secring.gpg
 
+  setup-git-credentials:
+    steps:
+      - run:
+          name: Setup Git config
+          command: |
+            git config user.email "ops@revenuecat.com"
+            git config user.name "revenuecat-ops"
```
Comment (aboedo) [inline .circleci/config.yml:67]: we should move these to env variables

File: .circleci/config.yml [hunk 2]
```diff
@@ -58,6 +58,34 @@ commands:
...
+
+  install-gems:
+    steps:
+      - restore_cache:
+          key: 1-gems-{{ checksum "Gemfile.lock" }}
+      - run: bundle check || bundle install --path vendor/bundle
+      - save_cache:
+          key: 1-gems-{{ checksum "Gemfile.lock" }}
+          paths:
+            - vendor/bundle
```
Comment (aboedo) [inline .circleci/config.yml:87]: we really have to improve the duplication of code for this kind of thing and fastlane. we should just set up a separate repo for them and set it up as a git submodule or something. I'll add it to the backlog

File: fastlane/Fastfile [hunk 3]
```diff
@@ -72,26 +72,50 @@ platform :android do
...
+#       ],
+#       properties: {
+#         "signing.keyId" => ENV['GPG_SIGNING_KEY_ID_NEW'],
+#         "signing.password" => ENV['GPG_SIGNING_KEY_PW_NEW'],
+#         "signing.secretKeyRingFile" => "/home/circleci/.gnupg/secring.gpg",
+#         "SONATYPE_NEXUS_USERNAME" => ENV['SONATYPE_NEXUS_USERNAME'],
+#         "SONATYPE_NEXUS_PASSWORD" => ENV['SONATYPE_NEXUS_PASSWORD'],
+#         "RELEASE_SIGNING_ENABLED" => true,
+#       }
+#     )
```
Comment (aboedo) [inline fastlane/Fastfile]: commented code

File: .circleci/config.yml (hunk 1 above)
Comment (vegaro) [reply .circleci/config.yml:67]: oopsies yes

File: .circleci/config.yml (hunk 2 above)
Comment (vegaro) [reply .circleci/config.yml:87]: We can use orbs for CircleCI and Fastlane I am not sure how it works but totally agree!

**Review by aboedo — APPROVED** [review]

**Comment by aboedo** [comment]
awesome

**Pull request merged**